mitodl/mitxpro
# Track enrollment company

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #187
Closes #300 

#### What's this PR do?
- Adds foreign key to `Company` in `ProgramEnrollment` and `CourseRunEnrollment` models, plus a method to determine what that company is.
- Filters `CourseEnrollment` by user and run_ids, not just run_ids, when determining if the user is enrolled in a course or not.

#### How should this be manually tested?
- Manually add a `BasketItem` to your `Basket`.
- Create a 100% off promo coupon that applies to your `BasketItem` product, and assign a company for it, via `../ecommerce/admin/coupons`
- Go to `../checkout`, enter the coupon code, and submit.
- Verify that the `CourseRunEnrollment` (or `ProgramEnrollment`) has the correct company assigned.
